webapp/src/selectors/index.js

```diff
@@ -1,10 +1,24 @@
+import {getConfig} from 'mattermost-redux/selectors/entities/general';
+
 import {id as pluginId} from '../manifest';
 
-const getPluginState = (state) => state['plugins-' + pluginId] || {};
+const getPluginState = (state) => state[`plugins-${pluginId}`] || {};
 
 const getAuthenticationResult = (state) => getPluginState(state).authenticationResult;
 
-export default {
-    getAuthenticationResult,
+export const getPluginServerRoute = (state) => {
+    const config = getConfig(state);
+
+    let basePath = '';
+    if (config && config.SiteURL) {
+        basePath = new URL(config.SiteURL).pathname;
+
+        if (basePath && basePath[basePath.length - 1] === '/') {
+            basePath = basePath.substr(0, basePath.length - 1);
+        }
+    }
+
+    return basePath;
 };
 
+export default {getAuthenticationResult};
```

Comment:
Can we instead export `getAuthenticationResult` like below, and adjust imports of the function to expect it to be exported in this manner?

```suggestion
export const getAuthenticationResult = (state) => getPluginState(state).authenticationResult;
```

Reply:
Removed